frontend/src/components/Dashboard.jsx

```diff
@@ -86,546 +86,580 @@ const DashBoardPage = styled.div`
-        toast.error("Invalid Address provided");
-      } else {
-        const PaymasterContract = getPaymasterContract(chainId);
-        const checkIfAdded = await PaymasterContract.check(
-          user.address,
-          whiteListAddress
-        );
-        if (checkIfAdded) {
-          toast.error("The Address has been already added to the whitelist");
-        } else {
-          const encodedData = PaymasterContract.interface.encodeFunctionData(
-            "addToWhitelist",
-            [whiteListAddress]
-          );
-          const txHash = await window.ethereum.request({
-            method: "eth_sendTransaction",
-            params: [
-              {
-                from: user.address, // The user's active address.
-                to: PaymasterContract.address, // Required except during contract publications.
-                data: encodedData,
-              },
-            ],
-          });
-          toast.loading(
-            (t) => (
-              <TransactionSentToast
-                txHash={txHash}
-                t={t}
-                blockExplorerLink={networks[chainId].blockExplorerLink}
-              />
-            ),
-            {
-              icon: "👏",
-            }
-          );
-        }
-      }
-      setIsLoading(false);
-    } catch (err) {
-      console.error(err);
-      toast.error("Something went wrong on MetaMask");
-      setIsLoading(false);
-    }
-  };
+	const handleWhitelistSubmit = async () => {
+		try {
+			setIsLoading(true);
+			if (!ethers.utils.isAddress(whiteListAddress)) {
+				toast.error("Invalid Address provided");
+			} else {
+				const PaymasterContract = getPaymasterContract(chainId);
+				const checkIfAdded = await PaymasterContract.check(
+					user.address,
+					whiteListAddress
+				);
+				if (checkIfAdded && !removeWhitelist) {
+					toast.error("The Address has been already added to the whitelist");
+					setIsLoading(false);
+					return;
+				}
+				if (!checkIfAdded && removeWhitelist) {
+					toast.error("The Address has not been added to the whitelist");
+					setIsLoading(false);
+					return;
+				}
+				const encodedData = PaymasterContract.interface.encodeFunctionData(
+					removeWhitelist ? "removeFromWhitelist" : "addToWhitelist",
+					[whiteListAddress]
+				);
+				const txHash = await window.ethereum.request({
+					method: "eth_sendTransaction",
+					params: [
+						{
+							from: user.address, // The user's active address.
+							to: PaymasterContract.address, // Required except during contract publications.
+							data: encodedData,
+						},
+					],
+				});
+				toast.loading(
+					(t) => (
+						<TransactionSentToast
+							txHash={txHash}
+							t={t}
+							blockExplorerLink={networks[chainId].blockExplorerLink}
+						/>
+					),
+					{
+						icon: "👏",
+					}
+				);
+			}
+			setIsLoading(false);
+		} catch (err) {
+			console.error(err);
+			toast.error("Something went wrong on MetaMask");
```

Comment:
As above, it could be any Browser-based wallet

Reply:
ohh okay